src/audio/component.c

```diff
@@ -29,20 +29,60 @@ DECLARE_SOF_UUID("component", comp_uuid, 0x7c42ce8b, 0x0108, 0x43d0,
 
 DECLARE_TR_CTX(comp_tr, SOF_UUID(comp_uuid), LOG_LEVEL_INFO);
 
-static const struct comp_driver *get_drv(uint32_t type)
+static const struct comp_driver *get_drv(struct sof_ipc_comp *comp)
 {
 	struct comp_driver_list *drivers = comp_drivers_get();
 	struct list_item *clist;
 	const struct comp_driver *drv = NULL;
 	struct comp_driver_info *info;
+	struct sof_ipc_comp_new_ext *comp_ext;
+	uint32_t comp_ext_size;
 	uint32_t flags;
 
 	irq_local_disable(flags);
 
+	/* do we have extended data ? */
+	if (!comp->ext_data_length)
+		goto comp_type_match;
+
+	comp_ext = (struct sof_ipc_comp_new_ext *)
+		   ((uint8_t *)comp + comp->hdr.size -
+		    comp->ext_data_length);
```

Comment:
@keyonjie Do we have a check for "hdr.size<ext_data_length" somewhere? If not, we do need one. Otherwise a  misbehaving driver can send IPC with ext_data_length larger than hdr.size and FW code will access data outside the IPC payload.

Reply:
@kv2019i good point, we don't have that yet, let me add this.